Feature/vue frontend

eklipse18 opened the pull request:
Merge the frontend implemented in Vue.
Todo:
- Clean up UI and the code.

eklipse18 committed:
implement a basic frontend with editor

bensgilbert committed:
install required dependencies

bensgilbert committed:
frontend additions

bensgilbert committed:
quick fix :)

bensgilbert committed:
border size consistency

deepsource-autofix[bot] committed:
Format code with prettier

eklipse18 committed:
remove node_modules

eklipse18 committed:
basic vue frontend, very dirty

Commit:
border consistency fix

deepsource-autofix[bot] committed:
Format code with prettier

eklipse18 committed:
use a more popular theme

LukeHankey reviewed (commented):
Looks good. I personally prefer the previous theme but I don't think choosing the theme is anything major at this point :p

Commit:
Border size consistency fix #2 :)

I missed a line of code before

eklipse18 committed:
add attribution

eklipse18 committed:
clean up css and add themes!!!

eklipse18 committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

Vthechamp22 commented:
I just need to handle empty room codes now

Evorage0 commented:
There needs to be a way to enter the username.

eklipse18 committed:
clean code, error message

error on incorrect room code

eklipse18 committed:
switch icon library & loading icon

Now you join the room after a timeout of 3 seconds. This mocks the time during which the user connects to the room.

deepsource-autofix[bot] committed:
Format code with prettier

eklipse18 committed:
add username input

eklipse18 committed:
forgot to remove fontawesome imports

eklipse18 committed:
basic readme for installation

deepsource-autofix[bot] committed:
Format code with prettier

Evorage0 commented:
Not ready yet, I need to finish up the web socket JS :)

LukeHankey commented on frontend/src/components/Home.vue:
```diff
@@ -3,16 +3,14 @@ import { ref, watch } from "vue";
-const show_modal = ref(false);
+const code = ref("");
+const username = ref("");
 const difficulty = ref(0);
```
Can you default the difficulty to 1

LukeHankey commented:
Front end looks really nice. Can we have the collaborators section show the username as in "Luke - You" as other people might have the same username.

Perhaps connect the websocket in a different PR. This looks good to get the frontend setup visually. Adding logic would be in a different scope

Vthechamp22 commented:
> Perhaps connect the websocket in a different PR. This looks good to get the frontend setup visually. Adding logic would be in a different scope

Yeah, I agree. I think let's merge this PR after I make these couple fixes instead of waiting for client JS to be finished.

eklipse18 committed:
set default difficulty to 1 and fix close icon

eklipse18 committed:
show username and add ability to pass down props to Room

eklipse18 committed:
Add validation for username field

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey reviewed (approved):
Looks good. ~Think you just need to run prettier on the files to pass the GHA.~ Seems prettier runs automatically by the bot. Just the few fixes on DeepSource to pass then

bensgilbert committed:
Frontend websocket integration

A hacky solution to interfacing with Vue.js, potentially code crashing :)

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
Merge branch 'main' into feature/vue-frontend

LukeHankey committed:
test

Vthechamp22 commented:
Do we need to satisfy deepsource before merging?

LukeHankey commented:
> Do we need to satisfy deepsource before merging?

Ideally, yes. It gives some really useful tips for better code quality

bensgilbert committed:
WebSocket support continued

deepsource-autofix[bot] committed:
Format code with prettier

Vthechamp22 commented:
@Evorage0 do you mind taking up the issues on deepsource? because most of it is from your part I think and I don't want to meddle with it because it may have unwanted consequences.

bensgilbert committed:
deepsource issue recognition + fix

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
websocket scope redefinition

LukeHankey commented on frontend/src/assets/js/main.js:
```diff
@@ -12,11 +12,19 @@ function generateRoomCode(length = 4) {
 let collaborators = [];
 let code = "";
 let connected = false;
-let websocket;
```
Might still need this as the `window.handleContentChange` requires the websocket.

LukeHankey commented on frontend/src/assets/js/main.js:
```diff
@@ -12,11 +12,19 @@ function generateRoomCode(length = 4) {
   editor = e.getModel();
+  editor.setValue(code);
+
+  var list = document.getElementById("collabul");
```
`const list = ...`

LukeHankey commented on frontend/src/assets/js/main.js:
```diff
@@ -76,38 +82,43 @@ function connect(username, roomCode, difficulty, hackyObject) {
-          collaborator.id !== message.data.id;
+          return collaborator.id !== message.data.id;
         });
         var list = document.getElementById("collabul");
```
Move line 91 above the switch statement and declare it as const. Can remove line 80 with this as well.

LukeHankey commented on frontend/src/assets/js/main.js:
```diff
@@ -76,38 +82,43 @@ function connect(username, roomCode, difficulty, hackyObject) {
+          return collaborator.id !== message.data.id;
         });
         var list = document.getElementById("collabul");
         var elem = document.getElementById(
```
Commenting here because I can't comment on line 81. Also move `var elem = ...` to above the switch statement and declare this one as `let`. It is only used in the connect and disconnect so they shouldn't interfere with each other ( I think)

kappq replied:
However, we can actually get the element only when it's created (basically when the connect event is sent).

LukeHankey replied:
Guess this makes sense within it's scope as you don't want to be creating the list element on each request

bensgilbert committed:
fixes

bensgilbert committed:
Removed self as collaborator

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
prevent redundent message

eklipse18 committed:
two more themes; temporary commit

eklipse18 committed:
add matching themes

bensgilbert committed:
Updated frontend 🥳

deepsource-autofix[bot] committed:
Format code with prettier

Commit:
Update Room.vue

Commit:
Fixed collaborator filter method

Commit:
Fixed connect event

bensgilbert committed:
Fixed code syncing

bensgilbert committed:
Fixed replace event

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
Merge branch 'main' into feature/vue-frontend

kappq committed:
Deepsource fixes (#50)

* Add custom status codes (#42)

* Add custom status codes

* Rename errors.py to codes.py

* Fix imports after module rename

This renames .errors to .codes.

* Remove useless argument

Removes the status code from RoomNotFound.

* Separate into different files

* Fix import after file separation

Rename .codes to .errors.

* Import StatusCode from codes

* More modifiers (#29)

New modifiers:
  Boolean swap - change True to False and False to True
  Condition destroyer - replace == with = in if conditions
  Type chaos - mix keywords like int, str, bool
  Bracket hell - add or remove some brackets

* Data models (#39)

* Add models for event types

* Remove room_code from DisconnectData

* Add username to DisconnectData

* Add error event type

* Rename ConnectionData to ConnectData

* Add validator to Event

* Add status_code to Event

* Remove old event data classes

* Rename cls to self

Configures flake8 to ignore N805.

* Remove useless print statement

* Convert received data to Event

* Move ConnectionManager methods

* Add ConnectionManager.connect method

This method handles the connection type directly in the ConnectionManager.

* Convert ConnectionManager.connect to a static method

* Make _update_code_cache public

This also sets the line length for isort to 119.

* Make EventType a subclass of str

This helps with JSON serialization.

* Convert data type to Event

This also changes the docstrings of the modified functions.

* Update code cache tests

Updates the tests because _update_code_cache was made public.

* Match field order with docstring

* Add data attribute to RoomNotFoundError

* Edit Event validator to accept EventData

* Add difficulty validator

* Separate EventRequest and EventResponse

EventRequest is the data sent from the client to the server, EventResponse is the data sent from the server to the client. It allows to easily handle the status codes.

* Fix difficulty type annotation

* Fix EventRequest and EventResponse docstrings

* Remove useless docstrings

* Fix RuntimeError

This avoids that the client sends a message to itself after he disconnects.

* feat: Add dev tools (#49)

* feat: Add more dev tools for development

* ci: Add flake8 plugins to flake8 pre-commit

* fix: flake8 errors fixed

* Use const for never reassigned variables

* Add default case for switch statement

* Change single character variable name

* Format code with prettier

* Adds error handling on receive and create (#47)

* fix: TypeError was raised if an incorrect data type (non-json) was sent

* fix: Adds error handling to client.receive() for incorrect input types

New error codes (4002, 4003, 4004) to be sent back to the client and then disconnected.
Frontend will have to check if any of these error codes are sent back during a live game by checking that the event type is not connect

* feat: Add new error: RoomAlreadyExistsError

* Adds error handling for new error and corrects types

* Adds JSONDecodeError if no data is sent at all

* Remove VALIDATION_ERROR code as not used

* removes unnecessary await/async from ConnectionManager.connect

* Remove whitespace from StatusCodes

* fix grammar

* kappa changes

* fix event being None if an error occurs

Only disconnect if event is None, raised by either a WebsocketDisconnect or a RuntimeError
Return a valid EventRequest object in any other error case with default values which won't modify any code

* feat: Adds a converter for the modified code (#51)

* feat: modify return structure

* test: Fix tests for output

* Change modifications detection

* remove print statement

* feat: Adds  the code modifier converter

* remove the deepdiff requirement

* fix flake8 errors

* Remove main guard

* Remove useless 'value' variable

Co-authored-by: Kappa <72104527+kappq@users.noreply.github.com>

* Project structure (#46)

* Separate Client and ConnectionManager in different files

* Change RoomData and ActiveRooms data structures

* Import from server instead of current package

* Use list as condition

This uses the list as a condition instead of checking if the length of the list is 0.

* feat: Broadcast buggy code (#35)

* Import modifiers to main

* feat: Adds the modification to the manager class and broadcasts buggy data

* docs: reword docs for modify param

* Transform code from string into list of strings for use with each modfiier

* Introduce a new EventType (SEND_BUGS)

The frontend should periodically send this event to broadcast bugs back to the client

* Fix black and isort compatability

* Clean up logic for broadcasting bugs

* Remove the last '\n' as too many get added. Copy _list_of_lines

There is a copy of _list_of_lines for file_contents otherwise that and modified_contents reference the same object

* test: Fix tests

* Resolve Luke comments

* Remove unused variables

* Fix invalid variable declaration

* Change var for const

* Format code with prettier

* feat: Add Event handler (#55)

* Adds the username to the Client class

* Consistently sends back an EventRequest on receive

We now handle the WebsocketDisconnect and RuntimeError in the EventHandler class so we no longer need to return None

* Update typing for data param in broadcast to accept any subclass of EventData

* Rename data instance attribute to response to reduce ambiguity

* Add the EventHandler class

* refactor main to incorporate the EventHandler

* self.connection > self.manager

* feat: Sync on join (#56)

* Adds the username to the Client class

* Consistently sends back an EventRequest on receive

We now handle the WebsocketDisconnect and RuntimeError in the EventHandler class so we no longer need to return None

* Update typing for data param in broadcast to accept any subclass of EventData

* Rename data instance attribute to response to reduce ambiguity

* Add the EventHandler class

* refactor main to incorporate the EventHandler

* Removes the connect method as EventHandler takes care of this now

* Update docs to be more accurate

* Update type for collaborators in SyncData

* Adds event handling logic for the sync event

* Fixed return type bug

Client.send expects an EventResponse (to include the event type) but the ConnectionManager.broadcast takes in and sent the EventData

* fix type issues with both type checkers

* Removed SEND_BUGS event till it's reworked

* forgot last assert statement

* Ignore node modules (#57)

* Ignore some deepsource warnings

* Format code with prettier

* spell skipcq correctly

* Add documentation

* Last function documented. No more deepsource warnings!

* Format code with prettier

* Format code with prettier

* No self.connection. Renamed to manager

* Send back sync on initial connects to connect to room

* deepsource fixes

* Format code with prettier

* Clear up invalid ref

Co-authored-by: Montarex23 <41803365+Montarex23@users.noreply.github.com>
Co-authored-by: Luke <68163698+LukeHankey@users.noreply.github.com>
Co-authored-by: deepsource-autofix[bot] <62050782+deepsource-autofix[bot]@users.noreply.github.com>
Co-authored-by: Luke Hankey <lukehankey@gmail.com>

LukeHankey committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

LukeHankey committed:
fix tests

LukeHankey committed:
fix merge conflicts

bensgilbert committed:
multiline fix + room code display

bensgilbert committed:
fixed multiline delete

bensgilbert committed:
Added support for lowercase room codes

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
set eol sequence for consistency

LukeHankey commented on frontend/src/components/Room.vue:
```diff
@@ -0,0 +1,197 @@
+});
+
+/**
+ * Function to conver the editor lines to index positions.
```
`convert`

eklipse18 committed:
fix exit icon

eklipse18 committed:
remove broken imports

eklipse18 committed:
add keys in v-for iteration

eklipse18 committed:
move username input above room code

eklipse18 committed:
remove room code check when creating room

eklipse18 committed:
satisfy deepsource

eklipse18 committed:
rework username input

deepsource-autofix[bot] committed:
Format code with prettier

eklipse18 committed:
fix CRLF and LF bug

bensgilbert committed:
EOL fix?

bensgilbert committed:
added timer and room info

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
added timings for room

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

eklipse18 committed:
Merge branch 'feature/monaco-themes' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

eklipse18 committed:
remove forced white color of title

bensgilbert committed:
fixed monaco theme select

bensgilbert committed:
fixed client collaborator list updates

bensgilbert committed:
unequivocally added support for lowercase roomcodes

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
unequivocally added support for lowercase roomcodes #2

bensgilbert committed:
stopped client disconnect after leaving room

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
stoped self being listed as collaborator

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

LukeHankey commented on server/events.py:
```diff
@@ -61,7 +61,13 @@ def valid_difficulty(cls, value, values):  # noqa: U100
+    """The data of a disconnection event.
+
+    Fields:
+        user: A list of users that are disconnecting
+    """
```
`user (optional):`

bensgilbert committed:
removed old unused file

eklipse18 committed:
add error handling

deepsource-autofix[bot] committed:
Format code with prettier

eklipse18 committed:
remove accidental bug

bensgilbert committed:
fixed second error not going away

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
Added room owner sync

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

bensgilbert committed:
Updates rooms code upon client sync

LukeHankey commented on server/event_handler.py:
```diff
@@ -127,23 +137,38 @@ async def __call__(self, request: EventRequest) -> bool:
+                if self.client.id != self.room.owner_id:
+                    return
+                
+                connect_data = cast(SyncData, event_data)
```
Why is it called `connect_data`?

LukeHankey commented on server/event_handler.py:
```diff
@@ -127,23 +137,38 @@ async def __call__(self, request: EventRequest) -> bool:
+                # Broadcast to every client (including sender) a sync event
+                response = EventResponse(
+                    type=EventType.SYNC,
+                    data=SyncData(code=cast(SyncData, event_data).code, collaborators=collaborators, time=time),
```
The cast here is not needed as the above `connect_data` is the same thing.

LukeHankey commented on server/events.py:
```diff
@@ -69,10 +76,12 @@ class SyncData(EventData):
     Fields:
         code: The code that already exists in the room.
         collaborators: The list of users that already collaborate in the room.
+        time: The elapsed time since the creation of the room.
```
`collaborators (optional):`
`time (optional):`

bensgilbert committed:
added eval requests?

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
Readded self as collaborator

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

bensgilbert committed:
added code evaluation modal

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
Merge branch 'main' into feature/vue-frontend

LukeHankey committed:
Event handler cleanup (#63)

* Cleans up duplicate code

* Removes empty whitespace

bensgilbert committed:
handle client sync event

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey reviewed (changes requested):
Just comments left to review

bensgilbert committed:
Added code evaluation UI for room owner

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey commented on server/events.py:
```diff
@@ -78,11 +78,13 @@ class SyncData(EventData):
         code: The code that already exists in the room.
         collaborators: The list of users that already collaborate in the room.
         time: The elapsed time since the creation of the room.
+        owner: The id of the owner of the room.
```
Why make it optional? This data could be sent with all sync responses and requests, no? Same with all of these optional fields.

bensgilbert committed:
Added support for all collaborator eval

bensgilbert committed:
Send sync before eval

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
fix evaluate event issues and ensure line wrapping for return code

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
Optional fields should be required. We have the data at each point to send to the clients

LukeHankey committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

LukeHankey committed:
Remove unnecessary casting

LukeHankey committed:
Fix typo

LukeHankey committed:
Edit docstring for optional field value

bensgilbert committed:
custom scrollbar

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

deepsource-autofix[bot] committed:
Format code with prettier

bensgilbert committed:
Fixed client disconnect on wrong room code

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

bensgilbert committed:
better error formatting

LukeHankey committed:
removed unused theme

LukeHankey committed:
sort imports

LukeHankey committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

LukeHankey committed:
remove console statements

LukeHankey committed:
Fix documentation deepsource warnings

deepsource-autofix[bot] committed:
Format code with prettier

LukeHankey committed:
Merge branch 'main' into feature/vue-frontend

LukeHankey committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

bensgilbert committed:
ignored TS error

bensgilbert committed:
Merge branch 'feature/vue-frontend' of https://github.com/Vthechamp22/kindly-kappa into feature/vue-frontend

bensgilbert committed:
ignored TS error
